pkg/loop/internal/relayer/pluginprovider/contractreader/contract_reader.go

```diff
@@ -196,6 +196,44 @@ func (c *Client) GetLatestValue(ctx context.Context, readIdentifier string, conf
 	return DecodeVersionedBytes(retVal, reply.RetVal)
 }
 
+func (c *Client) GetLatestValueWithHeadData(ctx context.Context, readIdentifier string, confidenceLevel primitives.ConfidenceLevel, params, retVal any) (*types.Head, error) {
+	_, asValueType := retVal.(*values.Value)
```

Comment:
Are you definitely trying to coerce this to a pointer to a Value (as opposed to just a Value?); Value is an interface.

Reply:
discussed offline